saleor/dashboard/product/views.py

```diff
@@ -73,3 +90,193 @@ def post(self, request, *args, **kwargs):
+    if request.method == 'POST':
+        stock.delete()
+        messages.success(request, _('Deleted stock'))
+        success_url = request.POST['success_url']
+        return redirect(success_url)
+    ctx = {'product': product, 'stock': stock}
+    return TemplateResponse(
+        request, 'dashboard/product/stock_confirm_delete.html', ctx)
+
+
+@staff_member_required
+def product_image_edit(request, product_pk, img_pk=None):
+    product = get_object_or_404(Product, pk=product_pk)
+    if img_pk:
+        product_image = get_object_or_404(product.images, pk=img_pk)
+        title = product_image.image.name
+    else:
+        product_image = ProductImage(product=product)
+        title = _('Add image')
+    form = ProductImageForm(request.POST or None, request.FILES or None,
+                            instance=product_image)
+    if form.is_valid():
+        product_image = form.save()
+        if img_pk:
+            msg = _('Updated image %s') % product_image.image.name
+        else:
+            msg = _('Added image %s') % product_image.image.name
+        messages.success(request, msg)
+        success_url = request.POST['success_url']
+        return redirect(success_url)
+    else:
+        if form.errors:
+            messages.error(request, _('Your submitted data was not valid - '
+                                      'please correct the errors below'))
+    ctx = {'product': product, 'product_image': product_image, 'title': title,
+           'form': form}
+    return TemplateResponse(
+        request, 'dashboard/product/product_image_form.html', ctx)
+
+
+@staff_member_required
+def product_image_delete(request, product_pk, img_pk):
+    product = get_object_or_404(Product, pk=product_pk)
+    product_image = get_object_or_404(product.images, pk=img_pk)
+    if request.method == 'POST':
+        product_image.delete()
+        messages.success(
+            request, _('Deleted image %s') % product_image.image.name)
+        success_url = request.POST['success_url']
+        return redirect(success_url)
+    ctx = {'product': product, 'product_image': product_image}
+    return TemplateResponse(
+        request, 'dashboard/product/product_image_confirm_delete.html', ctx)
+
+
+@staff_member_required
+def variant_edit(request, product_pk, variant_pk=None):
+    product = get_object_or_404(Product.objects.select_subclasses(),
+                                pk=product_pk)
+    variant_cls = get_variant_cls(product)
+    form_initial = {}
+    if variant_pk:
+        variant = get_object_or_404(product.variants.select_subclasses(),
+                                    pk=variant_pk)
+        title = variant.name
+    else:
+        variant = variant_cls(product=product)
+        title = _('Add variant')
+
+    form_cls = get_variant_form(product)
+    form = form_cls(request.POST or None, instance=variant,
+                    initial=form_initial)
+    attributes_form = VariantAttributesForm(request.POST or None,
```

Comment:
`attribute_form`, `VariantAttributeForm` should be singular just like you say "file manager" and not "files manager".